doc/dune-files.rst

```diff
@@ -787,6 +792,34 @@ A directory that is ignored will not be eagerly scanned by Dune. Any
 will be treated as raw data. It is however possible to depend on files
 inside ignored sub-directories.
 
+.. _include_subdirs:
+
+include_subdirs
+---------------
+
+The ``include_subdirs`` is used to control how dune considers
+sub-directories of the current directory. By default, it considers
+them as independent directories. However, by adding ``(include_subdirs
+<mode>)`` to your ``dune`` file, dune will assume that the
+sub-directories of the current directory are part of the same group of
+directories. In particular, dune will scan all these directories at
+once when looking for OCaml/Reason files. This allows you to split a
+library between several directories.
+
+Sub-directories are included recursively, however the recursion will
+stop when encountering a sub-directory that:
+
+- is part of a different project (for instance when vendoring projects)
+- contains ``(include_subdirs true)`` or ``(include_subdirs)``
```

Comment:
That should be `(include_subdirs unqualified)`

Reply:
& the short syntax has been removed if I understand correctly.

Reply:
Yep. Given that it's a new thing, it seems better to make things explicit